rdr_service/workflow_management/nph/sms_validation.py

```diff
@@ -0,0 +1,68 @@
+from rdr_service.services.ancillary_studies.nph_incident import create_nph_incident
+
+
+class SmsValidator:
+    def __init__(self, file_path, recipient):
+        self.file_path = file_path
+        self.recipient = recipient
+
+    def validate_bmi_value(
+        self, row: dict[str, str], max_length_allowed: int = 5
+    ) -> None:
+        """If bmi str length is greater than 5 chars (including the decimal), raise value error and send slack alert."""
+        bmi_str_len = len(row.get("bmi"))
+        if bmi_str_len > max_length_allowed:
+            msg = (
+                f"Pull list Validation Error: BMI value exceeds the max length allowed of {max_length_allowed} "
+                f"and contains {bmi_str_len} characters for sample id {row.get('sample_id')}."
+                f"Check the data at: {self.file_path}."
+            )
+            create_nph_incident(slack=True, message=msg)
+            raise ValueError(
+                f"BMI str length of {bmi_str_len} exceeds the limit of {max_length_allowed} characters."
+            )
```

Comment:
Made a comment concerning the expected length, but with this you can get a manifest that can have 1000 rows and this will alert up to the `max num of times` if the bmi length is too long. I would say its better to aggregate the alert in this case but might be better to extend the length of the value expected.

Reply:
This will only send 1 slack alert for each pull list it tries to ingest - as soon as it finds the first occurrence of bmi string length greater than 5, it will terminate, won't write to the `sms_sample` table, and send us an alert, so we can look at the file, and report back to the client.

Josh found out about this last week from the client that they would like BMI values to be a maximum of 5 characters.